### PR TITLE
k-textbox

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

* `core-textbox` -> `k-textbox`
* Minor clean up of textbox props, still just wrapping the keen-ui textbox.
* No more placeholders, always use the floating labels approved by @khangmach. Need to update guidelines.
* Class-enroll-page uses the k-textbox for the search filter, but it should not be using it. Replaced with `k-filter` in #1944

## TODO

- [ ] Clean up guidelines. 

127 0 0 1-8000-style_guide 1

### CODE REVIEW

File: kolibri/core/assets/src/views/k-textbox/index.vue [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+         * Emits input event with new value
+         */
+        this.$emit('input', this.currentText);
+      },
+      reset() {
+        this.$refs.textbox.reset();
+      },
+      emitKeydown(e) {
+        /**
+         * Emits keydown event with event
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/k-textbox/index.vue:128]: Maybe just "Emits keydown event"?

File: kolibri/plugins/style_guide/assets/src/views/content/text-fields/example.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  <div>
+    <k-textbox
+      v-model="model1"
+      label="Sample text field"
+      :disabled="false"
+      :required="true"
+      :invalid="isInvalid"
+      invalid-text="The input is invalid"
+      :autofocus="true"
```
Comment (indirectlylit) [inline kolibri/plugins/style_guide/assets/src/views/content/text-fields/example.html:11]: Setting this to `true` causes the style guide page to auto-scroll down when it's loaded.

File: kolibri/core/assets/src/views/k-textbox/index.vue [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * Whether or not required
+       */
+      required: {
+        type: Boolean,
+        default: false,
+      },
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/k-textbox/index.vue:63]: actually, I think we decided not to use the html `required` attribute, and to always use JS validation instead

**Comment by indirectlylit** [comment]
see https://github.com/learningequality/kolibri/issues/1666 for a case where we should be using JS instead of `required`

**Comment by indirectlylit** [comment]
see also this super minor issue that mildly infuriates me: https://github.com/learningequality/kolibri/issues/1626 :)

**Comment by christianmemije** [comment]
https://www.youtube.com/watch?v=jQ4C0aQlww4

Will address requested changes in upcoming PR.

**Comment by rtibbles** [comment]
It was labeled as 'ready for review' and it was approved, so I thought I was safe to assume it was ready for merge.

**Comment by christianmemije** [comment]
> see also this super minor issue that mildly infuriates me: #1626 :)
Since we will use the floating label by default, that change in border width will no longer be apparent.

**Comment by indirectlylit** [comment]
> Since we will use the floating label by default, that change in border width will no longer be apparent.

I don't _think_ that's true when the textbox has user-entered text in it

**Pull request merged**